daemon/cmd/kube_proxy_replacement.go

```diff
@@ -53,23 +53,28 @@ func initKubeProxyReplacementOptions() (strict bool) {
 		}
 	}
 
+	probesManager := probes.NewProbeManager()
+
 	if option.Config.KubeProxyReplacement == option.KubeProxyReplacementDisabled {
-		log.Infof("Auto-disabling %q, %q, %q, %q, %q features",
+		log.Infof("Auto-disabling %q, %q, %q, %q features",
 			option.EnableNodePort, option.EnableExternalIPs,
-			option.EnableHostReachableServices, option.EnableHostPort,
-			option.EnableSessionAffinity)
+			option.EnableHostReachableServices, option.EnableHostPort)
 
 		disableNodePort()
 		option.Config.EnableHostReachableServices = false
 		option.Config.EnableHostServicesTCP = false
 		option.Config.EnableHostServicesUDP = false
-		option.Config.EnableSessionAffinity = false
+
+		if option.Config.EnableSessionAffinity {
```

Comment:
Instead of duplicating the check, maybe you can move the existing check into a function with a `strict` param, and then invoke the function from here with `strict=false`?

Reply:
done